compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -124,8 +124,14 @@ fn create_mappings<'tcx>(
     let body_span = hir_info.body_span;
 
     let source_file = source_map.lookup_source_file(body_span.lo());
-    use rustc_session::RemapFileNameExt;
-    let file_name = Symbol::intern(&source_file.name.for_codegen(tcx.sess).to_string_lossy());
+
+    use rustc_session::{config::RemapPathScopeComponents, RemapFileNameExt};
+    let file_name = Symbol::intern(
+        &source_file
+            .name
+            .for_scope(tcx.sess, RemapPathScopeComponents::DEBUGINFO)
```

Comment:
This should also be MARCO. See https://github.com/rust-lang/rust/pull/122450#discussion_r1528386619.

Reply:
Hum, I'm pretty sure I did; I must have forgot this in the fight to fix the test.

```suggestion
            .for_scope(tcx.sess, RemapPathScopeComponents::MACRO)
```